src/uast.cc

```diff
@@ -200,9 +204,21 @@ void UastIteratorFree(UastIterator *iter) {
   }
 }
 
+UastIterator *UastIteratorNewWithTransformer(const Uast *ctx, void *node,
+                                             TreeOrder order, void*(*transform)(void*)) {
+
+  UastIterator *iter = UastIteratorNewBase(ctx, node, order);
```

Comment:
Would be nice to have a few `assert(ctx)` and alike to simplify the debugging once we go multithreaded

Reply:
Nice idea.